### PR TITLE
improve offline handling

### PR DESCRIPTION
**Pull request by hobbyquaker**
* Add new property _cached that is set if values are received from the gateways cache and not from the device
* handle subdevice heartbeats (needed for subdevice online/offline detection)

### CODE REVIEW

**Comment by hobbyquaker** [comment]
ready to merge now :)

File: src/lib/subdevice.js [hunk 1]
```diff
@@ -27,15 +27,36 @@ class Subdevice extends events.EventEmitter {
...
+    this._cached = Boolean(state.cached)
+
+    if (!state.cached && this._offline !== false) {
       this._offline = false
       this.emit('online')
     }
     this._rearmWatchdog()
   }
 
+  getCached () {
```
Comment (marvinroger) [inline src/lib/subdevice.js]: Can you please rename this to `isCached` and `isOffline`?

Comment (hobbyquaker) [reply src/lib/subdevice.js]: sure, done.

**Comment by marvinroger** [comment]
@hobbyquaker thanks, again! 😉

**Pull request merged**